crates/sui-json-rpc/src/governance_api.rs

```diff
@@ -451,6 +453,79 @@ pub struct ValidatorExchangeRates {
     pub rates: Vec<(EpochId, PoolTokenExchangeRate)>,
 }
 
+/// Backfill missing rates for some epochs due to safe mode. If a rate is missing for epoch e,
+/// we will use the rate for epoch e-1 to fill it.
+/// Rates returned are in descending order by epoch.
+fn backfill_rates(
+    rates: Vec<(EpochId, PoolTokenExchangeRate)>,
+) -> Vec<(EpochId, PoolTokenExchangeRate)> {
+    if rates.is_empty() {
+        return rates;
+    }
+
+    let min_epoch = *rates.iter().map(|(e, _)| e).min().unwrap();
```

Comment:
It's probably easier to sort it first before processing.
But otherwise this code looks good to me.

Reply:
I'm not sure if sorting it first will make things easier because then we'd have to keep track of the index we are at separately and insert the missing entry, etc. The logic may get messier. Unless I'm missing something.

Reply:
You just iterate the sorted vector. No need for min/max/find calls.